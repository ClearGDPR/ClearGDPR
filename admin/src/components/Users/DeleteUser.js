import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-responsive-modal';

import Loader from '../core/cards/dashboard/Loader';

const DeleteUser = props => (
  <Modal open={props.isOpen} showCloseIcon={false} onClose={() => {}} center>
    {!props.isLoading ? (
      <React.Fragment>
        <p>Are you sure you want to delete this user?</p>
        <p>
          <button
            className="ui-action btn"
            onClick={e => {
              e.preventDefault();
              props.onCancel();
            }}
          >
            Cancel
          </button>
          <button
            className="ui-action btn"
            onClick={e => {
              e.preventDefault();
              props.onConfirm();
            }}
          >
            OK
          </button>
        </p>
      </React.Fragment>
    ) : (
      <Loader />
    )}
  </Modal>
);

DeleteUser.propTypes = {
  onConfirm: PropTypes.func,
  onCancel: PropTypes.func,
  isOpen: PropTypes.bool,
  isLoading: PropTypes.bool
};

export default DeleteUser;
