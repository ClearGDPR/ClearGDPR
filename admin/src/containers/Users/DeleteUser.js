import React from 'react';
import PropTypes from 'prop-types';

import { UsersConsumer } from './UsersContext';
import DeleteUser from 'components/Users/DeleteUser';

export class DeleteUserContainer extends React.Component {
  static propTypes = {
    userId: PropTypes.number.isRequired,
    deleteUser: PropTypes.func.isRequired,
    onClose: PropTypes.func,
    isOpen: PropTypes.bool,
    isLoading: PropTypes.bool
  };

  deleteUser = () => {
    this.props.deleteUser(this.props.userId).then(this.props.onClose);
  };

  render() {
    return (
      <DeleteUser
        onConfirm={this.deleteUser}
        onCancel={this.props.onClose}
        isOpen={this.props.isOpen}
        isLoading={this.props.isLoading}
      />
    );
  }
}

export default props => (
  <UsersConsumer>
    {({ deleteUser, isLoading }) => (
      <DeleteUserContainer {...props} isLoading={isLoading} deleteUser={deleteUser} />
    )}
  </UsersConsumer>
);
