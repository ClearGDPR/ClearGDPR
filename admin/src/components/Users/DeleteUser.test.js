import React from 'react';
import { shallow, mount } from 'enzyme';

import DeleteUser from './DeleteUser';

const setup = propOverrides => {
  const props = Object.assign(
    {
      isOpen: true,
      onConfirm: jest.fn(),
      onCancel: jest.fn()
    },
    propOverrides
  );
  const component = shallow(<DeleteUser {...props} />);
  const mounted = mount(<DeleteUser {...props} />);

  return { props, component, mounted };
};

describe('(Component) Delete User', () => {
  it('should render correctly when closed', async () => {
    const { component } = setup({ isOpen: false });
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when open', async () => {
    const { component } = setup({ isOpen: true });
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to false', async () => {
    const { component } = setup({ isLoading: false });
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to true', async () => {
    const { component } = setup({ isLoading: true });
    expect(component).toMatchSnapshot();
  });

  it('should call onCancel when Cancel button clicked', async () => {
    const onCancel = jest.fn();
    const { mounted } = setup({ onCancel });

    const button = mounted.find('button').at(0);
    button.simulate('click');
    expect(onCancel).toHaveBeenCalled();
  });

  it('should call onConfirm when Confirm button clicked', async () => {
    const onConfirm = jest.fn();
    const { mounted } = setup({ onConfirm });

    const button = mounted.find('button').at(1);
    button.simulate('click');
    expect(onConfirm).toHaveBeenCalled();
  });
});
