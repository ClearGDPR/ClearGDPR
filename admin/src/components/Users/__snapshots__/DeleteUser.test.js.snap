// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Delete User should render correctly when closed 1`] = `
<Modal
  animationDuration={500}
  center={true}
  classNames={Object {}}
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "overlay": "styles_overlay__CLSq-",
      "overlayCenter": "styles_overlayCenter__YHoO7",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={true}
  closeOnOverlayClick={true}
  onClose={[Function]}
  onEntered={null}
  onEscKeyDown={null}
  onExited={null}
  onOverlayClick={null}
  open={false}
  showCloseIcon={false}
  styles={Object {}}
>
  <React.Fragment>
    <p>
      Are you sure you want to delete this user?
    </p>
    <p>
      <Component
        onClick={[Function]}
        text="Cancel"
      />
       
      <Component
        onClick={[Function]}
        text="OK"
      />
    </p>
  </React.Fragment>
</Modal>
`;

exports[`(Component) Delete User should render correctly when open 1`] = `
<Modal
  animationDuration={500}
  center={true}
  classNames={Object {}}
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "overlay": "styles_overlay__CLSq-",
      "overlayCenter": "styles_overlayCenter__YHoO7",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={true}
  closeOnOverlayClick={true}
  onClose={[Function]}
  onEntered={null}
  onEscKeyDown={null}
  onExited={null}
  onOverlayClick={null}
  open={true}
  showCloseIcon={false}
  styles={Object {}}
>
  <React.Fragment>
    <p>
      Are you sure you want to delete this user?
    </p>
    <p>
      <Component
        onClick={[Function]}
        text="Cancel"
      />
       
      <Component
        onClick={[Function]}
        text="OK"
      />
    </p>
  </React.Fragment>
</Modal>
`;

exports[`(Component) Delete User should render correctly when the prop isLoading is set to false 1`] = `
<Modal
  animationDuration={500}
  center={true}
  classNames={Object {}}
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "overlay": "styles_overlay__CLSq-",
      "overlayCenter": "styles_overlayCenter__YHoO7",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={true}
  closeOnOverlayClick={true}
  onClose={[Function]}
  onEntered={null}
  onEscKeyDown={null}
  onExited={null}
  onOverlayClick={null}
  open={true}
  showCloseIcon={false}
  styles={Object {}}
>
  <React.Fragment>
    <p>
      Are you sure you want to delete this user?
    </p>
    <p>
      <Component
        onClick={[Function]}
        text="Cancel"
      />
       
      <Component
        onClick={[Function]}
        text="OK"
      />
    </p>
  </React.Fragment>
</Modal>
`;

exports[`(Component) Delete User should render correctly when the prop isLoading is set to true 1`] = `
<Modal
  animationDuration={500}
  center={true}
  classNames={Object {}}
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "overlay": "styles_overlay__CLSq-",
      "overlayCenter": "styles_overlayCenter__YHoO7",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={true}
  closeOnOverlayClick={true}
  onClose={[Function]}
  onEntered={null}
  onEscKeyDown={null}
  onExited={null}
  onOverlayClick={null}
  open={true}
  showCloseIcon={false}
  styles={Object {}}
>
  <Loader />
</Modal>
`;
